### PR TITLE
Add netbeans/idea/eclipse themes

### PR DESCRIPTION
**Pull request by ericdallo**
* add more themes
* add Cask file
* separate themes to another file

based on https://github.com/sebastiencs/company-box/pull/98

### CODE REVIEW

File: lsp-treemacs-themes.el [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    (treemacs-create-icon :file "typeparameter.png" :extensions (type-parameter) :fallback "-")
+    (treemacs-create-icon :file "template.png" :extensions (template) :fallback "-")))
+
+(treemacs-create-theme "Idea"
+  :icon-directory (f-join (f-dirname (or load-file-name buffer-file-name)) "icons/idea")
+  :config
+  (progn
+    (treemacs-create-icon :file "vscode/template.png" :extensions (template) :fallback "-")
+    (treemacs-create-icon :file "package.png" :extensions (unknown) :fallback "-")
+    ;; (treemacs-create-icon :file "misc.png" :extensions (text) :fallback "-")
```
Comment (ericdallo) [inline lsp-treemacs-themes.el:158]: @yyoncho why these were commented on `company-box` PR?

File: lsp-treemacs-themes.el [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    (treemacs-create-icon :file "event.png" :extensions (event) :fallback "-")
+    (treemacs-create-icon :file "operator.png" :extensions (operator) :fallback "-")
+    (treemacs-create-icon :file "typeparameter.png" :extensions (type-parameter) :fallback "-")
+    (treemacs-create-icon :file "template.png" :extensions (template) :fallback "-")))
+
+(treemacs-create-theme "Netbeans"
+  :icon-directory (f-join (f-dirname (or load-file-name buffer-file-name)) "icons/netbeans")
+  :config
+  (progn
+    (treemacs-create-icon :file "vscode/Namespace.png" :extensions (namespace) :fallback "-")
```
Comment (ericdallo) [inline lsp-treemacs-themes.el:124]: Should we use vscode(default) for these that don't have namespace?

File: lsp-treemacs-themes.el (hunk 1 above)
Comment (yyoncho) [reply lsp-treemacs-themes.el:158]: I think that those are missing, no?

File: lsp-treemacs-themes.el (hunk 2 above)
Comment (yyoncho) [reply lsp-treemacs-themes.el:124]: Either this or pick another icon from the theme. Not sure which one will look more consistent.

**Pull request merged**